### DISCUSSION TITLE
Missing Sentinel 2LA Data

### DISCUSSION THREAD

**Discussion by RobertBehling** [microsoft/PlanetaryComputer#165] [General]
Hi,
I retrieved Sentinel 2 data from planetary computer for a given AOI. The AOI intersects two MGRStiles (36KVC, 36KVD). This data is retrieved and merged. For the given two Sentinel2 tiles I got the following result for the date 2019-09-16:
<img src=https://user-images.githubusercontent.com/80827055/213713251-cf621c95-9b91-4078-bd6d-0e9723da3d36.png width="300"/>
The black data is not due to swath edges, it is along the swath, which is strange. I checked the data availablity on the [planetrary computer explorer](https://planetarycomputer.microsoft.com/explore?c=33.0456%2C-20.3052&z=8.54&v=2&d=sentinel-2-l2a&m=cql%3A7b3a63afa0d0056eb6aca88c29ce2929&r=Natural+color&s=false%3A%3A100%3A%3Atrue&sr=desc&ae=0) and got the same result:
<img src=https://user-images.githubusercontent.com/80827055/213715029-f4aec235-c0e2-4e8e-ae32-96497a607e2f.png width="300"/>  
So the data seems to be really missing.

I never experienced something like that before, so I checked also the data availablility on [ESA's OpenHub](https://scihub.copernicus.eu/dhus/#/home).
There, for the same MGRS tile id 36KVC, two S2 datasets are existing (which I also have not experienced so far):
<img src=https://user-images.githubusercontent.com/80827055/213715764-47da2f06-2601-4208-882d-d0e9426f5fff.png width="300"/>
<img src=https://user-images.githubusercontent.com/80827055/213715793-36c7874c-247d-4177-b6fc-be11ec3c3f47.png width="300"/>

Interestingly such behavior does not occur for other dates I've checked, e.g. 2019-06-08
<img src=https://user-images.githubusercontent.com/80827055/213716577-91c5e08b-7787-4103-b1b9-4441da5ef012.png width="300"/>

==> 1) What is the reason why they are missing on Planetary Computer?   
==> 2) Does anyone has an idea how often it occurs that for an individual MGRS tile more than one S2 dataset exists?

**Comment by TomAugspurger** [comment]
> ==> 1) What is the reason why they are missing on Planetary Computer?

Just to make sure I understand: you suspect there are some scenes missing from the Planetary Computer? Specifically, we have the "bottom" part (https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a/items/S2A_MSIL2A_20190916T073611_R092_T36KVC_20201004T134047), but are missing that top section? On https://finder.creodias.eu/, searching for the string S2A_MSIL2A_20190916T073611_N0213_R092_T36KVC_20201021T025647.SAFE
 should give you that scene that's apparently missing. If you can confirm my understanding, I can dig into things to see what's going on.

> ==> 2) Does anyone has an idea how often it occurs that for an individual MGRS tile more than one S2 dataset exists?

I'm not sure.

**Comment by RobertBehling** [comment]
Thanks for the prompt answer!
Yes, the upper part with the ID S2A_MSIL2A_20190916T073611_N0213_R092_T36KVC_20201021T025647.SAFE seems to be missing in Planetary Computer, but is present in [Creodias](https://finder.creodias.eu/)

**Reply by TomAugspurger** [reply]
Thanks. I've kicked of a job to grab those specific items, though it'll take some time to propagate through.

In the meantime, I'll keep investigating why the items weren't present in the first place and will do a larger audit for missing scenes.

**Comment by zanifooy** [comment]
Has the problem of missing data on planetary computer been resolved? I seem to have the same problem.

I am missing the following scenes on planetary computer which are available on other servers:

<Item id=S2A_11SQT_20211106_0_L2A>
<Item id=S2A_11SQT_20211103_0_L2A>
<Item id=S2B_11SQT_20211101_0_L2A>
<Item id=S2B_11SQT_20211029_0_L2A>
<Item id=S2A_11SQT_20211027_1_L2A>
<Item id=S2A_11SQT_20211027_0_L2A>
<Item id=S2A_11SQT_20211024_0_L2A>
<Item id=S2B_11SQT_20211022_0_L2A>

**Reply by TomAugspurger** [reply]
The majority of the reported issues have been fixed. We still have a project ongoing to get a full audit of our scenes compared to what's available through ESA.

I think your reply might have been cut off before you submitted the missing scenes. Feel free to post those and I'll take a look when I get a chance.

**Reply by zanifooy** [reply]
My apologies, here is the list: S2A_11SQT_20211106_0_L2A, S2A_11SQT_20211103_0_L2A, S2B_11SQT_20211101_0_L2A, S2B_11SQT_20211029_0_L2A, S2A_11SQT_20211027_1_L2A, S2A_11SQT_20211027_0_L2A, S2A_11SQT_20211024_0_L2A, S2B_11SQT_20211022_0_L2A
